wagtail/wagtailsearch/backends/elasticsearch.py

```diff
@@ -26,7 +26,7 @@ class ElasticsearchMapping(object):
         'DateField': 'date',
         'DateTimeField': 'date',
         'DecimalField': 'double',
-        'FileField': 'string',
+        'FileField': 'attachment',
```

Comment:
I like your solution, but it will mean that people will have to create their own class Document 
````python
class MyDocument(AbstractDocument):
    admin_form_fields = (
        'title',
        'file',
        'collection',
        'tags'
    )
    search_fields = CollectionMember.search_fields + [
        index.SearchField('title', partial_match=True, boost=10),
        index.SearchField('file', search_content=True),
        index.RelatedFields('tags', [
            index.SearchField('name', partial_match=True, boost=10),
        ]),
        index.FilterField('uploaded_by_user'),
    ]
````
and overwrite `settings.WAGTAILDOCS_DOCUMENT_MODEL` accordingly.


would that be fine with everybody?